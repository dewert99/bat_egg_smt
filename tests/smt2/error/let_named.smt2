(declare-const z Bool)
(define-const x Bool (or (! false :named y) unbound))
(define-const x Bool true)
(define-const y Bool false) ; these should work since the first command failed and nothing was bound
(define-const a Bool (! z :named a))