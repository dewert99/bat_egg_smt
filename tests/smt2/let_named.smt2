(declare-const a Bool)
(declare-const b Bool)
(declare-fun f (Bool) Bool)
(assert (and (! a :named x) (let ((x b)) x) (f x)))
(check-sat)
(get-model)
(assert (not (and a b (f a))))
(check-sat)