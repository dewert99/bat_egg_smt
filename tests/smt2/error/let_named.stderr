unknown identifier `unbound` at 2:45
(define-const x Bool (or (! false :named y) unbound))
                                            ^

the :named identifier `a` is already in scope at 5:34
(define-const a Bool (! z :named a))
                                 ^

